fix: Add missing timezone options for TimeRangeRule

gecolay opened the pull request:
### 1. Why is this change necessary?
- See "3. Describe each step to reproduce the issue or behaviour"

### 2. What does this change do, exactly?
* Added `timezone` options to `Shopware\Core\Framework\Rule\TimeRangeRule`
* Changed `Resources/app/administration/src/app/component/rule/condition-type/sw-condition-time-range/sw-condition-time-range.html.twig` to display the timezone options below the time range fields
* Changed `Resources/app/administration/src/app/component/rule/condition-type/sw-condition-time-range/index.js` to handle the new timezone options

### 3. Describe each step to reproduce the issue or behaviour.
- Configure a time range rule in the rule builder for e.g. the availability of a shipping method. e.g. 22:00 - 12:30 which would match 00:00 - 14:30 in Europe/Berlin right now with 2 hour offset
- Experience a daylight saving time change in your time zone
- The rule now no longer matches the correct time, as it was saved in UTC 22:00 - 12:30 before to match Europe/Berlin 00:00 - 14:30 which now after the daylight saving time change matches Europe/Berlin 23:00 - 13:30

### 4. Please link to the relevant issues (if any).
- /

### 5. Checklist

- [X] I have written tests and verified that they fail without my change
- [X] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [X] I have written or adjusted the documentation according to my changes
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I have read the contribution requirements and fulfilled them

larskemper commented on src/Administration/Resources/app/administration/src/app/component/rule/condition-type/sw-condition-time-range/index.js:
```diff
@@ -52,14 +63,36 @@ export default {
+
+    methods: {
+        createdComponent() {
+            this.timezoneOptions = Shopware.Service('timezoneService').getTimezoneOptions();
```
Wouldn't using a computed be a better approach here?

```
timezoneOptions() {
    return Shopware.Service('timezoneService').getTimezoneOptions();
},
```

gecolay replied:
This is how it is done in various places in the administration, but of course we can also have a different implementation for this component.

larskemper commented on src/Administration/Resources/app/administration/src/app/component/rule/condition-type/condition-type.spec.js:
```diff
@@ -9,13 +9,16 @@ import ConditionDataProviderService from 'src/app/service/rule-condition.service
     return conditionType.match(/^(?!_mocks)[a-z-]*(?<!\.spec)$/);
 });
 
+const timeZoneService = new TimezoneService();
```
It would be better to mock the timezone options here to ensure isolation

gecolay replied:
Changed it.

larskemper commented on src/Administration/Resources/app/administration/src/app/component/rule/condition-type/sw-condition-time-range/sw-condition-time-range.html.twig:
```diff
@@ -1,29 +1,57 @@
 {% block sw_condition_value_content %}
```
We need to discuss this with our design team to see if they're satisfied with the solution.

larskemper replied:
@gecolay We discussed and asked ourselves: what’s the specific use case for having two separate time-zone selectors? Could you clarify which specifcy need you have for that? From a design standpoint, we’d like to keep the rule-builder on a single row, so we’d prefer a single time-zone dropdown for both start and end times, positioned to the right of the time pickers.

larskemper commented on src/Administration/Resources/app/administration/src/app/component/rule/condition-type/sw-condition-time-range/sw-condition-time-range.scss:
```diff
@@ -5,6 +5,12 @@
     gap: 4px;
     grid-template-columns: 1fr 1fr;
 
+    > div {
```
use a more specific selector if possible

gecolay replied:
Changed it to a specific class

larskemper commented on src/Core/Framework/Rule/TimeRangeRule.php:
```diff
@@ -41,12 +46,18 @@ public function getConstraints(): array
-    private function extractTime(string $time, \DateTimeImmutable $now): \DateTimeInterface
+    private function extractTime(string $time, ?string $timezone, \DateTimeImmutable $now): \DateTimeInterface
     {
+        if ($timezone) {
```
What happends if only one timezone is filled? Do we want to allow this? What is the default timezone?

gecolay replied:
Yes, you can specify only a single timezone if you wish. This is intended.
Allowing empty time zones and different time zones gives you the best way to fully customize for your needs.

The default timezone is like everywhere in shopware the default timezone of the system.

gecolay committed:
Add missing timezone option for TimeRangeRule

gecolay commented:
@larskemper I adjusted it to keep only a single timezone option. I updated the admin layout like you requested it.
<img width="1188" height="82" alt="image" src="https://github.com/user-attachments/assets/e3ecc220-f266-4b44-b839-3c89ad607b05" />

gecolay committed:
scss lint fix

larskemper committed:
Merge branch 'trunk' into add-timezone-to-time-range-rule
